scheduler-driver/src/test/java/gov/nasa/jpl/aerie/scheduler/PrioritySolverTest.java

```diff
@@ -228,4 +231,48 @@ public void getNextSolution_coexistenceGoalOnActivityWorks() {
         .containsAtLeastElementsIn(expectedPlan.getActivitiesByTime());
   }
 
+
+  @Test
+  public void testCardGoalWithApplyWhen(){
+    var planningHorizon = h;
+
+    final var fooMissionModel = SimulationUtility.getFooMissionModel();
+    Problem problem = new Problem(fooMissionModel, planningHorizon, new SimulationFacade(
+        planningHorizon,
+        fooMissionModel), SimulationUtility.getFooSchedulerModel());
+    final var activityType = problem.getActivityType("ControllableDurationActivity");
+
+    //act at t=1hr and at t=2hrs
+    problem.setInitialPlan(makePlanA12(problem));
+
+    //and the goal window in between [0, 50 min[ so the activities already present in the plan can't count towards satisfying the goal
+    final var goalWindow = new Windows(false).set(List.of(
+        Interval.between(Duration.of(0, Duration.SECONDS), Duration.of(50, Duration.MINUTE))
+    ), true);
+
+    //unsatisfiable because 10 occurences asked when only one can be placed
```

Comment:
Is this comment still accurate?

Reply:
No, it is not, will remove before merge, thanks.